hooks/read-package-hook/src/createPeerDependencyPatcher.ts

```diff
@@ -44,6 +46,22 @@ export function createPeerDependencyPatcher (
 
       pkg.peerDependencies![peerName] = currentVersions.join(' || ')
     }
+
+    let overrides: ReturnType<typeof parseOverrides>
+    try {
+      overrides = parseOverrides(peerDependencyRules.allowedVersions ?? {})
+    } catch (e) {
+      const selector = (e as PnpmError)?.message?.match(/".*"/)?.[0] ?? ''
+      throw new PnpmError('INVALID_ALLOWED_VERSION_SELECTOR', `Cannot parse the ${selector} selector in the allowedVersions`)
+    }
+
+    const [peerDepsOverrides] = partition(({ parentPkg }) => parentPkg !== undefined && parentPkg.name === pkg.name, overrides)
```

Comment:
it doesn't look like a partition is needed here. It looks like a filter is enough